File: java/src/test/java/com/iteso/nintendo/CharizardTest.java
package com.iteso.nintendo;

import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.assertEquals;

/**
 * Created by rvillalobos on 2/24/18.
 */
public class CharizardTest {
    Charizard charizard;

    @Before
    public void setUp(){
        charizard = new Charizard();
    }

    /** Main attack test. */
    @Test
    public void mainAttackTest() {
        assertEquals("Attacking opponent with Small Fire causing a damage of 3", charizard.mainAttack());
    }

    /** Second attack test. */
    @Test
    public void secondAttackTest() {
        assertEquals("Attacking opponent with Quick Attack causing a damage of 5", charizard.secondAttack());
    }

    /** Defense test */
    @Test
    public void defendTest() {
        assertEquals("Defending attack, damage caused is 0 new HP is 77", charizard.defend(0));
    }

    /** Name test */
    @Test
    public void getNameTest() {
        assertEquals("Charizard", charizard.getName());
    }

    /** Evolve test */
    @Test
    public void evolveTest() {
        assertEquals(null, charizard.evolve());
    }

    /** Type test */
    @Test
    public void typeTest() {
        assertEquals("fire", charizard.getType());
    }

    /** Evolvution test */
    @Test
    public void hasEvolutionTest() {
        assertEquals(true, charizard.hasEvolution());
    }

    /** Main Damage test */
    @Test
    public void getMainAttackDamageTest() {
        assertEquals(3, charizard.getMainAttackDamage());
    }

    /** Second Damage test */
    @Test
    public void getSecondAttackDamageTest() {
        assertEquals(5, charizard.getSecondAttackDamage());
    }

    /** Hit points test */
    @Test
    public void getHitPointsTest() {
        assertEquals(77, charizard.getHitPoints());
    }

    /** Defense multiplier test */
    @Test
    public void getDefenseMultiplierTest() {
        assertEquals(0.3, charizard.getDefenseMultiplier(), 0.0);
    }
}

File: python/src/unittest/python/CharizardTest.py
import unittest
from Charizard import *


class CharizardTest(unittest.TestCase):
    def setUp(self):
        self.charizard = Charizard()

    def mainAttackTest(self):
        self.assertEquals(
            "Attacking opponent with Small Fire causing a damage of 3",
            self.charizard.mainAttack())

    def secondAttackTest(self):
        self.assertEquals(
            "Attacking opponent with Quick Attack causing a damage of 5",
            self.charizard.secondAttack())

    def defendTest(self):
        self.assertEquals(
            "Defending attack, damage caused is 0 new HP is 77",
            self.charizard.defend(0))

    def getNameTest(self):
        self.assertEquals("Charizard", self.charizard.getName())

    def evolveTest(self):
        self.assertEquals(None, self.charizard.evolve())

    def typeTest(self):
        self.assertEquals("fire", self.charizard.getType())

    def hasEvolutionTest(self):
        self.assertEquals(True, self.charizard.isHasEvolution())


    def getMainAttackDamageTest(self):
        self.assertEquals(3, self.charizard.getMainAttackDamage())

    def getSecondAttackDamageTest(self):
        self.assertEquals(5, self.charizard.getSecondAttackDamage())

    def getDefenseMultiplierTest(self):
        self.assertEquals(0.3, self.charizard.getDefenseMultiplier())

    def getHitPointsTest(self):
        self.assertEquals(77, self.charizard.getHitPoints())